### PR TITLE
Fix ranger melee attack due to changes in the latest patch

### PR DESCRIPTION
**Pull request by chobobobo**
moving to a hostile used to be an implicit attack, it no longer is.

### CODE REVIEW

File: creep.action.defending.js [hunk 1]
```diff
@@ -47,7 +47,10 @@ action.run = {
             }
             if( range < 3 ){
                 let direction = creep.target.pos.getDirectionTo(creep);
-                if( direction ) creep.move(direction);
+                if( direction ) {
+                    creep.move(direction);
+                    creep.attack(creep.target)
```
Comment (karlthepagan) [inline creep.action.defending.js]: I forgot to set `creep.attacking = true` with this recommendation

Comment (chobobobo) [inline creep.action.defending.js]: However, creep.attacking was not set previously, so my thinking is that it doesn't need to be backported?

**Comment by chobobobo** [comment]
squashed it so I can cherrypick more easily.

File: creep.action.defending.js [hunk 2]
```diff
@@ -47,7 +47,12 @@ action.run = {
             }
             if( range < 3 ){
                 let direction = creep.target.pos.getDirectionTo(creep);
-                if( direction ) creep.move(direction);
+                if( direction ) {
+                    creep.move(direction);
+                    if (range === 1) {
+                        creep.attacking = creep.attack(creep.target) == OK;
```
Comment (Spedwards) [inline creep.action.defending.js]: (\*screams\*) _THREE EQUAL SIGNS!_

**Pull request merged**